### ISSUE TITLE
On Linux, clang is not selected as a compiler.

### ISSUE THREAD

**Issue by funcodeio** [centaurean/density#38]
Hello,

I tried to build density under Linux environment, but I cannot choose clang as a compiler even if clang is installed properly. It seems that following trick in premake5 does not work no longer on Linux.

```
if os.execute("clang -v") == 0 then
  premake.cc  = 'clang'
  premake.cxx = 'clang++'
elseif os.execute("gcc -v") == 0 then
  premake.cc  = 'gcc'
  premake.cxx = 'g++'
```

Any result Makefile does not contain something like "CC=clang" or "CXX=clang++"

```
~/workspace/density/density/build$ ./premake5 gmake  && grep clang * -r
Ubuntu clang version 3.6.0-2ubuntu1~trusty1 (tags/RELEASE_360/final) (based on LLVM 3.6.0)
Target: x86_64-pc-linux-gnu
Thread model: posix
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.6
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.6.4
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8.2
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.9.1
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.6.4
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8.2
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9.1
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8
Candidate multilib: .;@m64
Selected multilib: .;@m64
git version 1.9.1
Building configurations...
Running action 'gmake'...
Generating Makefile...
Generating spookyhash.make...
Generating density.make...
Generating benchmark.make...
Done.
Binary file premake5 matches
premake5.lua:if os.execute("clang -v") == 0 then
premake5.lua:	premake.cc  = 'clang'
premake5.lua:	premake.cxx = 'clang++
```

If it is not specified, CC indicates cc, CXX indicates c++. They are symbolic links to gcc, and g++, respectively by default. Therefore, although you recommend to use clang, clang is not selected and gcc is used by default.

My Linux is  Ubuntu 14.04
$ uname -a
Linux neo-work 3.13.0-24-generic #47-Ubuntu SMP Fri May 2 23:30:00 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

**Comment by gpnuma** [comment]
Hello
Thanks for your feedback ! You're right it doesn't work properly on linux, nor on OS/X actually for that matter. I didn't notice it because clang is by default the main compiler on that platform.
I'll go ahead and ask the premake5 devs what's the proper way of "favoring" clang if it's available.
Thanks again

**Comment by gpnuma** [comment]
Fixed in acda00cb7f525cf4466419494ba0dd8271cc27dc.
I re-released 0.12.3 with this fix included.
